### PR TITLE
Fix and add examples for storage_profile resource and data source

### LINKED ISSUE #76: vRA Select & Use Storage Profiles
Disk and Storage profile syntax and formatting documentation is non existent. As with the networks, please give us an example usage to select and use storage profiles

### PR DESCRIPTION
**Pull request by dmettem**
Fixes #76.

Signed-off-by: Deepak Mettem <dmettem@vmware.com>

### CODE REVIEW

File: examples/storage_profile/aws_with_ebs/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# storage_profile example for AWS with EBS device type
+
+This is an example on how to create an AWS Storage Profile with EBS device type. 
+
+## Getting Started
+
+There are variables which need to be added to terraform.tfvars. The first are for connecting to the vRealize Automation (vRA) endpoint. The others are names of cloud_account, and region that already setup in vRA.
```
Comment (Prativa20) [inline examples/storage_profile/aws_with_ebs/README.md:7]: that `are` already setup in vRA?

File: examples/storage_profile/aws_with_ebs/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+To facilitate adding these variables, a sample tfvars file can be copied first:
+```shell
+cp terraform.tfvars.sample terraform.tfvars
+```
+
+Follow these examples for setting up specific cloud accounts:
+
+* Setup [cloud\_account\_aws](../../cloud_account_aws/README.md)
+* Setup [cloud\_account\_azure](../../cloud_account_azure/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/aws_with_ebs/README.md:24]: this link says page not found

File: examples/storage_profile/aws_with_instance_store/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# storage_profile example for AWS with instance store device type
+
+This is an example on how to create an AWS Storage Profile with instance store device type. 
+
+## Getting Started
+
+There are variables which need to be added to terraform.tfvars. The first are for connecting to the vRealize Automation (vRA) endpoint. The others are names of cloud_account, and region that already setup in vRA.
```
Comment (Prativa20) [inline examples/storage_profile/aws_with_instance_store/README.md:7]: same

File: examples/storage_profile/aws_with_ebs/README.md (hunk 2 above)
Comment (Prativa20) [inline examples/storage_profile/aws_with_ebs/README.md:24]: this link says page not found

File: examples/storage_profile/aws_with_instance_store/README.md (hunk 3 above)
Comment (Prativa20) [inline examples/storage_profile/aws_with_instance_store/README.md:7]: same as above

File: examples/storage_profile/aws_with_instance_store/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+To facilitate adding these variables, a sample tfvars file can be copied first:
+```shell
+cp terraform.tfvars.sample terraform.tfvars
+``` 
+
+Follow these examples for setting up specific cloud accounts:
+
+* Setup [cloud\_account\_aws](../../cloud_account_aws/README.md)
+* Setup [cloud\_account\_azure](../../cloud_account_azure/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/aws_with_instance_store/README.md:24]: same

File: examples/storage_profile/azure_with_managed_disks/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# storage_profile example for Azure with managed disk storage type
+
+This is an example on how to create an Azure Storage Profile with managed disk storage type. 
+
+## Getting Started
+
+There are variables which need to be added to terraform.tfvars. The first are for connecting to the vRealize Automation (vRA) endpoint. The others are names of cloud_account, and region that already setup in vRA.
```
Comment (Prativa20) [inline examples/storage_profile/azure_with_managed_disks/README.md:7]: same

File: examples/storage_profile/azure_with_managed_disks/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+To facilitate adding these variables, a sample tfvars file can be copied first:
+```shell
+cp terraform.tfvars.sample terraform.tfvars
+```
+
+Follow these examples for setting up specific cloud accounts:
+
+* Setup [cloud\_account\_aws](../../cloud_account_aws/README.md)
+* Setup [cloud\_account\_azure](../../cloud_account_azure/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/azure_with_managed_disks/README.md:24]: please verify if it is same as above

File: examples/storage_profile/azure_with_unmanaged_disks/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# storage_profile example for Azure with unmanaged disk storage type
+
+This is an example on how to create an Azure Storage Profile with unmanaged disk storage type. 
+
+## Getting Started
+
+There are variables which need to be added to terraform.tfvars. The first are for connecting to the vRealize Automation (vRA) endpoint. The others are names of cloud_account, and region that already setup in vRA.
```
Comment (Prativa20) [inline examples/storage_profile/azure_with_unmanaged_disks/README.md:7]: same

File: examples/storage_profile/azure_with_unmanaged_disks/README.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+To facilitate adding these variables, a sample tfvars file can be copied first:
+```shell
+cp terraform.tfvars.sample terraform.tfvars
+```
+
+Follow these examples for setting up specific cloud accounts:
+
+* Setup [cloud\_account\_aws](../../cloud_account_aws/README.md)
+* Setup [cloud\_account\_azure](../../cloud_account_azure/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/azure_with_unmanaged_disks/README.md:25]: same

File: examples/storage_profile/vsphere_with_first_class_disk/README.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
+# storage_profile example for vSphere with standard disk
+
+This is an example on how to create a vSphere Storage Profile with first class disk. 
+
+## Getting Started
+
+There are variables which need to be added to terraform.tfvars. The first are for connecting to the vRealize Automation (vRA) endpoint. The others are names of cloud_account, and region that already setup in vRA.
```
Comment (Prativa20) [inline examples/storage_profile/vsphere_with_first_class_disk/README.md:7]: same

File: examples/storage_profile/vsphere_with_first_class_disk/README.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+* `storage_policy_name` - The name of the vSphere fabric storage policy
+
+To facilitate adding these variables, a sample tfvars file can be copied first:
+```shell
+cp terraform.tfvars.sample terraform.tfvars
+```
+
+Follow these examples for setting up specific cloud accounts:
+
+* Setup [cloud\_account\_aws](../../cloud_account_aws/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/vsphere_with_first_class_disk/README.md:25]: Do we need this information for vsphere storage profile?

File: examples/storage_profile/vsphere_with_first_class_disk/README.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+* Setup [cloud\_account\_azure](../../cloud_account_azure/README.md)
+* Setup [cloud\_account\_gcp](../../cloud_account_gcp/README.md)
+* Setup [cloud\_account\_vmc](../../cloud_account_vmc/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/vsphere_with_first_class_disk/README.md:28]: Do we need this information for vsphere storage profile?

File: examples/storage_profile/azure_with_unmanaged_disks/README.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+* Setup [cloud\_account\_gcp](../../cloud_account_gcp/README.md)
+* Setup [cloud\_account\_vmc](../../cloud_account_vmc/README.md)
+* Setup [cloud\_account\_vsphere](../../cloud_account_vsphere/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/azure_with_unmanaged_disks/README.md:28]: Do we need this for azure storage profile?

File: examples/storage_profile/aws_with_ebs/README.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+* Setup [cloud\_account\_gcp](../../cloud_account_gcp/README.md)
+* Setup [cloud\_account\_vmc](../../cloud_account_vmc/README.md)
+* Setup [cloud\_account\_vsphere](../../cloud_account_vsphere/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/aws_with_ebs/README.md:27]: azure link on line 24 says page not found

Do we need all this information for aws storage profile?

File: examples/storage_profile/aws_with_instance_store/README.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+* Setup [cloud\_account\_gcp](../../cloud_account_gcp/README.md)
+* Setup [cloud\_account\_vmc](../../cloud_account_vmc/README.md)
+* Setup [cloud\_account\_vsphere](../../cloud_account_vsphere/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/aws_with_instance_store/README.md:27]: Same as above

File: examples/storage_profile/azure_with_managed_disks/README.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+* Setup [cloud\_account\_gcp](../../cloud_account_gcp/README.md)
+* Setup [cloud\_account\_vmc](../../cloud_account_vmc/README.md)
+* Setup [cloud\_account\_vsphere](../../cloud_account_vsphere/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/azure_with_managed_disks/README.md:27]: Same as above

File: examples/storage_profile/azure_with_unmanaged_disks/README.md (hunk 12 above)
Comment (Prativa20) [inline examples/storage_profile/azure_with_unmanaged_disks/README.md:28]: Same

File: examples/storage_profile/vsphere_with_standard_disk/README.md [hunk 16]
```diff
@@ -0,0 +1,36 @@
+# storage_profile example for vSphere with standard disk
+
+This is an example on how to create a vSphere Storage Profile with standard disk type, independent and non-persistent disk mode. 
+
+## Getting Started
+
+There are variables which need to be added to terraform.tfvars. The first are for connecting to the vRealize Automation (vRA) endpoint. The others are names of cloud_account, and region that already setup in vRA.
```
Comment (Prativa20) [inline examples/storage_profile/vsphere_with_standard_disk/README.md:7]: same as above

File: examples/storage_profile/vsphere_with_standard_disk/README.md [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+```shell
+cp terraform.tfvars.sample terraform.tfvars
+```
+
+Follow these examples for setting up specific cloud accounts:
+
+* Setup [cloud\_account\_aws](../../cloud_account_aws/README.md)
+* Setup [cloud\_account\_azure](../../cloud_account_azure/README.md)
+* Setup [cloud\_account\_gcp](../../cloud_account_gcp/README.md)
+* Setup [cloud\_account\_vmc](../../cloud_account_vmc/README.md)
```
Comment (Prativa20) [inline examples/storage_profile/vsphere_with_standard_disk/README.md:28]: same as above

File: examples/storage_profile/aws_with_ebs/README.md (hunk 1 above)
Comment (dmettem) [reply examples/storage_profile/aws_with_ebs/README.md:7]: Done.

File: examples/storage_profile/aws_with_instance_store/README.md (hunk 3 above)
Comment (dmettem) [reply examples/storage_profile/aws_with_instance_store/README.md:7]: Done.

Comment (dmettem) [reply examples/storage_profile/aws_with_instance_store/README.md:7]: Done.

File: examples/storage_profile/azure_with_unmanaged_disks/README.md (hunk 7 above)
Comment (dmettem) [reply examples/storage_profile/azure_with_unmanaged_disks/README.md:7]: Done.

File: examples/storage_profile/vsphere_with_first_class_disk/README.md (hunk 9 above)
Comment (dmettem) [reply examples/storage_profile/vsphere_with_first_class_disk/README.md:7]: Done.

File: examples/storage_profile/azure_with_managed_disks/README.md (hunk 5 above)
Comment (dmettem) [reply examples/storage_profile/azure_with_managed_disks/README.md:7]: Done.

File: examples/storage_profile/vsphere_with_standard_disk/README.md (hunk 16 above)
Comment (dmettem) [reply examples/storage_profile/vsphere_with_standard_disk/README.md:7]: Done.

File: examples/storage_profile/vsphere_with_standard_disk/README.md (hunk 17 above)
Comment (dmettem) [reply examples/storage_profile/vsphere_with_standard_disk/README.md]: Done.

File: examples/storage_profile/azure_with_unmanaged_disks/README.md (hunk 12 above)
Comment (dmettem) [reply examples/storage_profile/azure_with_unmanaged_disks/README.md]: Done.

File: examples/storage_profile/azure_with_managed_disks/README.md (hunk 15 above)
Comment (dmettem) [reply examples/storage_profile/azure_with_managed_disks/README.md]: Done.

File: examples/storage_profile/aws_with_ebs/README.md (hunk 13 above)
Comment (dmettem) [reply examples/storage_profile/aws_with_ebs/README.md]: Done.

File: examples/storage_profile/aws_with_instance_store/README.md (hunk 14 above)
Comment (dmettem) [reply examples/storage_profile/aws_with_instance_store/README.md]: Done.

File: examples/storage_profile/vsphere_with_first_class_disk/README.md (hunk 11 above)
Comment (dmettem) [reply examples/storage_profile/vsphere_with_first_class_disk/README.md]: Done.

File: examples/storage_profile/README.md [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+
+This directory covers examples on how to create different kinds of Storage Profiles. 
+
+## Getting Started
+
+Follow these examples for setting up Storage Profiles:
+
+* Setup [aws storage_profile](aws_with_ebs/README.md) with EBS device type
+* Setup [aws storage_profile](aws_with_instance_store/README.md) with instance store device type
+* Setup [azure storage_profile](azure_with_managed_disks/README.md) with managed disks
```
Comment (Prativa20) [inline examples/storage_profile/README.md:11]: page not found

**Review by Prativa20 — APPROVED** [review]

**Pull request merged**